runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/util/BoundedQueueExecutor.java

```diff
@@ -73,7 +74,8 @@ public BoundedQueueExecutor(
           protected void beforeExecute(Thread t, Runnable r) {
             super.beforeExecute(t, r);
             synchronized (BoundedQueueExecutor.this) {
-              if (++activeCount >= maximumPoolSize && startTimeMaxActiveThreadsUsed == 0) {
+              if (++activeCount >= BoundedQueueExecutor.this.maximumPoolSize
```

Comment:
can you improve the unit test to catch this? there are separate tests for changing limit and calculating the time but one with both could catch this.

Reply:
done